presentation/common/src/presentation-common/content/Fields.ts

```diff
@@ -187,6 +217,47 @@ export class Field {
     });
   }
 
+  public static decompressFieldJSON(json: CompressedFieldJSON, classesMap: { [id: string]: CompressedClassInfoJSON }): FieldJSON | undefined {
+    if (isPropertiesField(json))
+      return this.decompressPropertiesFieldJSON(json, classesMap);
+
+    if (isNestedContentField(json))
+      return this.decompressNestedContentFieldJSON(json, classesMap);
+
+    return json;
+  }
+
+  private static decompressNestedContentFieldJSON(compressedNestedContentFieldJSON: CompressedNestedContentFieldJSON, classesMap: { [id: string]: CompressedClassInfoJSON }): NestedContentFieldJSON {
+    return {
+      ...compressedNestedContentFieldJSON,
+      contentClassInfo: { id: compressedNestedContentFieldJSON.contentClassInfo, ...classesMap[compressedNestedContentFieldJSON.contentClassInfo] },
+      nestedFields: compressedNestedContentFieldJSON.nestedFields.map((compressedFieldJSON) => this.decompressFieldJSON(compressedFieldJSON, classesMap)).filter((decompressedJson): decompressedJson is FieldJSON => !!decompressedJson),
+      pathToPrimaryClass: compressedNestedContentFieldJSON.pathToPrimaryClass.map((compressedInfoJSON) => SelectClassInfo.decompressRelatedClassInfoJSON(compressedInfoJSON, classesMap)),
+    };
+  }
+
+  private static decompressPropertiesFieldJSON(compressedPropertiesFieldJSON: CompressedPropertiesFieldJSON, classesMap: { [id: string]: CompressedClassInfoJSON }): PropertiesFieldJSON {
+    return {
+      ...compressedPropertiesFieldJSON,
+      properties: compressedPropertiesFieldJSON.properties.map((compressedPropertyJSON) => Field.decompressPropertyJSON(compressedPropertyJSON, classesMap)),
+    };
+  }
+
+  private static decompressPropertyJSON(compressedPropertyJSON: CompressedPropertyJSON, classesMap: { [id: string]: CompressedClassInfoJSON }): PropertyJSON {
+    return {
+      property: Field.decompressPropertyInfoJSON(compressedPropertyJSON.property, classesMap),
+      // eslint-disable-next-line deprecation/deprecation
+      relatedClassPath: compressedPropertyJSON.relatedClassPath.map((compressedInfoJSON) => SelectClassInfo.decompressRelatedClassInfoJSON(compressedInfoJSON, classesMap)),
+    };
+  }
+
+  private static decompressPropertyInfoJSON(compressedPropertyJSON: CompressedPropertyInfoJSON, classesMap: { [id: string]: CompressedClassInfoJSON }): PropertyInfoJSON {
+    return {
+      ...compressedPropertyJSON,
+      classInfo: {id: compressedPropertyJSON.classInfo, ...classesMap[compressedPropertyJSON.classInfo]},
+    };
+  }

```

Comment:
I'd move these from class scope to module scope as plain functions.

Reply:
Moved it.